core/iwasm/common/wasm_memory.c

```diff
@@ -274,7 +274,8 @@ wasm_runtime_validate_app_addr(WASMModuleInstanceCommon *module_inst_comm,
         goto fail;
     }
 
-    /* integer overflow check */
+    /* integer overflow check, and if memory boundary check enabled, app_offset
```

Comment:
Consider add a unchecked version of these functions for AOT, to improve performance (dont't need do if in each memory access operation)